src/geojson/Feature.cpp

```diff
@@ -0,0 +1,209 @@
+            break;
+        default:
+            type = FeatureType::None;
+    }
+
+    geometry = new_geometry;
+    bounding_box = new_bounding_box;
+    properties = new_properties;
+}
+
+Feature::Feature(std::vector<geojson::JSONGeometry> new_geometry_collection,
+                 std::vector<double> new_bounding_box,
+                 property_map new_properties) {
+    type = geojson::FeatureType::GeometryCollection;
+    geometry_collection = std::move(new_geometry_collection);
+    properties = std::move(new_properties);
+    bounding_box = std::move(new_bounding_box);
+}
+
+Feature::Feature(boost::property_tree::ptree &tree) {
+    if (tree.count("geometry") > 0) {
+        boost::property_tree::ptree geom = tree.get_child("geometry");
+        geometry = JSONGeometry::from_ptree(geom);
+
+        switch(geometry.get_type()) {
+            case JSONGeometryType::Point:
+                type = FeatureType::Point;
+                break;
+            case JSONGeometryType::LineString:
+                type = FeatureType::LineString;
+                break;
+            case JSONGeometryType::Polygon:
+                type = FeatureType::Polygon;
+                break;
+            case JSONGeometryType::MultiPoint:
+                type = FeatureType::MultiPoint;
+                break;
+            case JSONGeometryType::MultiLineString:
+                type = FeatureType::MultiLineString;
+                break;
+            case JSONGeometryType::MultiPolygon:
+                type = FeatureType::MultiPolygon;
+                break;
+            default:
+                type = FeatureType::None;
+        }
+    }
+    else if (tree.count("geometries") > 0) {
+        type = FeatureType::GeometryCollection;
+        for (auto &geom : tree.get_child("geometries")) {
+            geometry_collection.push_back(JSONGeometry::from_ptree(geom.second));
+        }
+    }
+
+    if (tree.count("bbox") > 0) {
+        for (auto &value : tree.get_child("bbox")) {
+            bounding_box.push_back(std::stod(value.second.data()));
+        }
+    }
+
+    if (tree.count("properties") > 0) {
+        for (auto &property : tree.get_child("properties")) {
+            // TODO: Add handling for nested objects by determining if property.second is another ptree
+            std::string key = property.first.data();
+            std::string value = property.second.data();
+
+            bool is_numeric = true;
+            bool is_real = true;
+            bool decimal_already_hit = false;
+
+            if (property.second.data() == "true") {
+                properties.emplace(key, JSONProperty(key, true));
+                continue;
+            }
+            else if (property.second.data() == "false") {
+                properties.emplace(key, JSONProperty(key, false));
+                continue;
+            }
+
+            for(int character_index = 0; character_index < value.length(); character_index++) {
```

Comment:
I would be inclined to move this character parsing to a private function by itself.